samples/cpp/multiview_calibration_sample.cpp

```diff
@@ -80,6 +110,28 @@ static void detectPointsAndCalibrate (cv::Size pattern_size, float pattern_scale
             {
                 success = cv::findCirclesGrid(img, pattern_size, corners, cv::CALIB_CB_ASYMMETRIC_GRID);
             }
+            else if (pattern_type == "charuco")
+            {
+                std::vector<int> ids; cv::Mat corners_sub;
+                detector->detectBoard(img, corners_sub, ids);
+                corners.create(board.size(), 2, CV_32F);
+                if (ids.size() < 4)
+                    success = false;
+                else {
+                    success = true;
+                    int head = 0;
+                    for (size_t i = 0; i < board.size(); i++) {
+                        if (head < ids.size() && ids[head] == i) {
+                            corners.at<float>(i, 0) = corners_sub.at<float>(head, 0);
+                            corners.at<float>(i, 1) = corners_sub.at<float>(head, 1);
+                            head++;
+                        } else {
+                            corners.at<float>(i, 0) = -1.;
+                            corners.at<float>(i, 1) = -1.;
+                        }
+                    }
```

Comment:
It does not work, if not all Charuco corners are detected/labeled correctly. corners array will contain (-1,-1) and it'll be used in calibration pipeline.

Reply:
I put it to be (-1, -1) on purpose, as internally, I would check whether the point is inside the image frame x in (0, width-1), y in (0, height-1). If not, this point will be discarded. I design the input of the method to be like this so that I only need to feed in a single copy of the objective points. Otherwise, I would need to have a copy of the objective function for every frame and for every camera